### PR TITLE
fix(readevents): fix next page handling

### LINKED ISSUE #28: Doing a .readEvents while providing next_page is broken
If you provide a next page option https://github.com/cronofy/cronofy-node/blob/master/src/index.js#L183 (which I get back from the api in a previous call)
It will still prefix the provided path with the cronofy host here https://github.com/cronofy/cronofy-node/blob/master/src/index.js#L28

I think the easiest solution is to just see if the next page options starts with `this.urls.api` and then cut it off.

### PR DESCRIPTION
**Pull request by AdriVanHoudt**
Closes #28

### CODE REVIEW

**Comment by gshutler** [comment]
While I can see this would work, it doesn't feel quite right.

I'm thinking if `this.urls.api + path` was moved to `_httpGet`, `_httpPost`, and `_httpDelete` then we could have:

```js
if (details.options.next_page) {
  return this. _httpCall('GET', details.options.next_page, details.options, details.callback);
} else {
  return this._httpGet('/v1/events', details.options, details.callback);
}
```

So you use `_httpCall` direct when you have the full URL, but `_httpGet` and friends are helpers so you only need to specify the path.

WDYT?

**Comment by AdriVanHoudt** [comment]
👍 nice suggestion
Would it make sense to omit the `next_page` from the options so it does not end up in the entity?
e.g. do `return this._httpCall('GET', details.options.next_page, details.options, details.callback, ['next_page']);`

**Comment by gshutler** [comment]
Yes, I think so. I don't think it will do any harm but as there's no point in sending it we shouldn't.

This needs the [`this.urls.api + path`](https://github.com/cronofy/cronofy-node/blob/03abd0b29520d21961fcb216d97140f5a753bead/src/index.js#L28) moving from `_httpCall` to `_httpGet` and friends like so:

```js
cronofy.prototype._httpGet = function(path, options, callback, optionsToOmit){
  return this._httpCall('GET', this.urls.api + path, options, callback, optionsToOmit);
}

cronofy.prototype._httpPost = function(path, options, callback, optionsToOmit){
  return this._httpCall('POST', this.urls.api + path, options, callback, optionsToOmit);
}

cronofy.prototype._httpDelete = function(path, options, callback, optionsToOmit){
  return this._httpCall('DELETE', this.urls.api + path, options, callback, optionsToOmit);
}

/* ... */

cronofy.prototype._httpCall = function(method, path, options, callback, optionsToOmit){
  var settings = {
    method: method,
    path: path,
    headers: {
      Authorization: 'Bearer ' + options.access_token,
      'Content-Type': 'application/json'
    },
    entity: _.omit(options, optionsToOmit || ['access_token'])
  };
```

**Comment by AdriVanHoudt** [comment]
Totally missed that 👌

File: src/index.js [hunk 1]
```diff
@@ -180,6 +180,10 @@ cronofy.prototype.profileInformation = function(){
 cronofy.prototype.readEvents = function(){
   var details = this._parseArguments(arguments, ["access_token"]);
 
+  if (details.options.next_page) {
+    return this._httpCall('GET', details.options.next_page, details.options, details.callback, ['access_token', 'next_page']);
+  }
+
   return this._httpGet(details.options.next_page || '/v1/events', details.options, details.callback);
```
Comment (gshutler) [inline src/index.js]: Could just be:

```js
this._httpGet('/v1/events', ...
```

As `details.options.next_page` will never be truthy at this point any more.

Comment (AdriVanHoudt) [inline src/index.js]: Clearly not awake haha sorry

**Comment by gshutler** [comment]
Now just need to work out the best way to handle the inevitable merge conflict with #30 😄

**Comment by AdriVanHoudt** [comment]
I'll rebase once you merge this, no sweat 💦

**Comment by gshutler** [comment]
Great, thanks!

**Pull request merged**